src/app/features/event/event-list/event-list.component.html

```diff
@@ -4,9 +4,49 @@
   </div>
   <ng-template #success>
     <app-cow-loader></app-cow-loader>
-    <ng-container *ngIf="eventList$ | async as events">
+    <mat-tab-group
+      class="event-list__tab-group"
+      [ngClass]="{ 'event-list__tab-group--hidden': isHidden }"
+      animationDuration="0ms"
+      (click)="toggleTab()"
+      (selectedTabChange)="openTab()">
+      <mat-tab label="Sortuj">
+        <menu class="menu-list" (click)="toggleTab()">
+          <li *ngFor="let sortOption of sortOptions">
+            <button (click)="sort(sortOption.value)">{{ sortOption.name }}</button>
+          </li>
+        </menu>
+      </mat-tab>
+      <mat-tab label="Filtruj">
+        <div class="event-list__tab" (click)="toggleTab()">
+          <p *ngFor="let filterOption of filterOptions; let i = index">
+            <mat-checkbox
+              [formControl]="formArray.controls[i]"
+              color="primary"
+              (change)="addfilter(filterOption.value)"
+              >{{ filterOption.name }}</mat-checkbox
+            >
+          </p>
+          <div class="event-list__button-container">
+            <button mat-flat-button color="primary" (click)="filter()">filtruj</button>
+            <button mat-flat-button color="warn" (click)="resetFilter()">resetuj</button>
+          </div>
+        </div>
+      </mat-tab>
+      <mat-tab label="Kategorie">
+        <menu class="menu-list" (click)="toggleTab()">
+          <li *ngFor="let category of categories$ | async">
+            <button (click)="chooseCategory(category.name)">{{ category.name }}</button>
+          </li>
+          <li class="menu-list__reset">
+            <button (click)="resetCategory()">Resetuj kategorię</button>
+          </li>
+        </menu>
```

Comment:
Czy tutaj nie powinien być jakis `ngIf="categories$ | async as categories"` i dopiero potem `*ngFor` skoro jest to zapytanie do bazy i coś może pójść nie tak?